src/__tests__/createReactContext.test.js

```diff
@@ -72,3 +78,19 @@ test('with provider', () => {
   wrapper.find('button').simulate('click');
   expect(wrapper).toMatchSnapshot('with provider - after click');
 });
+
+test('unsubscribes correct consumer', () => {
+  const wrapper = mount(
+    <ThemeToggler>
+      <Title>Hello World</Title>
+      <Title>Foo</Title>
+      <Title>Bar</Title>
+    </ThemeToggler>
+  );
+
+  expect(wrapper).toMatchSnapshot();
+  wrapper.setState({ hiddenChildren: [1] });
+  expect(wrapper).toMatchSnapshot();
+  wrapper.find('button').simulate('click');
+  expect(wrapper).toMatchSnapshot();
+})
```

Comment:
It's not really clear that this tests unsubscribing the right consumer. Instead of modifying the existing component, could you create a new one that wraps two different `<Title>`'s and hides one of them?

Reply:
Not sure what is the idea behind proposed tweaked test - I mean I'm not sure in what way would you see it tweaked.

This test is testing that correct component got unsubscribed (or rather that others are still subscribed). Each `<Title/>` has different children, so we can see if those components got updated. I agree though that this might not be as clear as it could be - mainly because it's a snapshot test.